### PR TITLE
fix: bundle ripgrep and fall back to it when VS Code's copy is not found

### PR DESCRIPTION
**Pull request by 0xMink**
## Summary

`search_files` and `list_files` fail with `Could not find ripgrep binary` on VS Code Insiders' newer staged-install layout, where program files moved into a per-build subfolder (microsoft/vscode#252063). `getBinPath` only checked four hardcoded paths under `vscode.env.appRoot`, none of which resolve on that layout.

This bundles a ripgrep binary inside the extension and uses it as a fallback:

- Add `@vscode/ripgrep` as a build devDependency.
- New `copyRipgrep` esbuild plugin copies the platform `rg` binary into `dist/bin/` so it ships inside the VSIX.
- `getBinPath` gains a final fallback to the bundled binary; VS Code's own copy is still preferred when present.
- CI asserts the binary is present in the packaged VSIX.

All three `getBinPath` callers (`search_files`, `list_files`, `file-search`) inherit the fix with no changes.

## Scope — PR 1 of 2

The extension ships a single universal VSIX, and `@vscode/ripgrep` installs only the build host's platform binary, so the released VSIX carries `rg` for the CI build platform only. That fully fixes that platform and is a non-regression elsewhere — on a platform whose binary is not bundled, the fallback simply finds nothing, exactly as before this change. A follow-up PR will move to platform-specific VSIXs so every platform ships a usable binary.

## Test plan

- [x] `vitest` — new `getBinPath` tests: fallback used when nothing exists under the app root; VS Code's own copy preferred over the bundled one; `undefined` when ripgrep exists nowhere.
- [x] `pnpm vsix` — the packaged VSIX contains `extension/dist/bin/rg`.
- [ ] Install the built VSIX on a VS Code Insiders machine with the staged-install layout and confirm `search_files` / `list_files` work.

## Summary by CodeRabbit

* **Bug Fixes**
  * Bundled a platform/architecture-specific ripgrep binary and added a fallback so search and file-listing work when ripgrep isn't found in some VS Code installs (fixes "Could not find ripgrep binary").
* **Tests**
  * Added coverage to ensure the extension prefers a VS Code-provided ripgrep when present, falls back to the bundled copy, and reports when none are available.
* **Chores**
  * Added a packaged-content check to validate the bundled ripgrep is included in release artifacts.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Zoo-Code-Org/Zoo-Code/pull/243?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/services/ripgrep/index.ts [hunk 1]
```diff
@@ -92,7 +106,8 @@ export async function getBinPath(vscodeAppRoot: string): Promise<string | undefi
 		(await checkPath("node_modules/@vscode/ripgrep/bin/")) ||
 		(await checkPath("node_modules/vscode-ripgrep/bin")) ||
 		(await checkPath("node_modules.asar.unpacked/vscode-ripgrep/bin/")) ||
-		(await checkPath("node_modules.asar.unpacked/@vscode/ripgrep/bin/"))
+		(await checkPath("node_modules.asar.unpacked/@vscode/ripgrep/bin/")) ||
+		((await fileExistsAtPath(bundledRgPath)) ? bundledRgPath : undefined)
```
[the remark this answers is not in the document]
Comment (0xMink) [reply src/services/ripgrep/index.ts:120]: Addressed in 838fb9107.

The bundled binary is now copied to and resolved from `dist/bin/<platform>-<arch>/` (e.g. `dist/bin/linux-x64/rg`). `bundledRgPath` includes the `${process.platform}-${process.arch}` segment, so on macOS it resolves to `dist/bin/darwin-arm64/rg` — which a Linux-built universal VSIX does not contain — and the fallback correctly yields nothing instead of an incompatible binary. A test was added asserting the path is platform/arch-scoped.

File: src/esbuild.mjs [hunk 2]
```diff
@@ -85,6 +85,27 @@ async function main() {
...
+					const { rgPath } = await import("@vscode/ripgrep")
+					if (!rgPath) {
+						throw new Error("[copyRipgrep] @vscode/ripgrep did not provide rgPath")
+					}
+					const rgDestDir = path.join(distDir, "bin")
+					fs.mkdirSync(rgDestDir, { recursive: true })
+					const rgDest = path.join(rgDestDir, path.basename(rgPath))
+					fs.copyFileSync(rgPath, rgDest)
+					fs.chmodSync(rgDest, 0o755)
+					console.log(`[copyRipgrep] Copied ${rgPath} to ${rgDest}`)
```
[the remark this answers is not in the document]
Comment (0xMink) [reply src/esbuild.mjs:109]: Addressed in 838fb9107.

`copyRipgrep` now writes the binary under `dist/bin/${process.platform}-${process.arch}/`, and `getBinPath` resolves the matching `<platform>-<arch>` segment at runtime. A binary built for one OS can no longer be picked up by the fallback on another — even though the VSIX is universal and `rg` shares a filename across Linux and macOS.

**Comment by proyectoauraorg** [comment]
Nice fix — preferring VS Code's own `rg` and falling back to a SHA512-pinned bundled binary is a clean way to handle the staged-install breakage. One QA/hardening thought: logging which `rg` path actually resolved (VS Code vs bundled vs platform) would make the resolved source visible in support threads and surface any unexpected fallback. Happy to help add resolver-order test coverage (VS Code → bundled → none) if useful. 🙌

**Review by proyectoauraorg — APPROVED** [review]
## Code Review: Bundle ripgrep fallback

**Overall Assessment: Well-designed solution with thorough testing. Approve with minor suggestions.**

### Strengths

1. **Clean fallback chain**: The `getBinPath()` now prefers VS Code's own ripgrep and falls back to the bundled copy — correct priority order.

2. **Platform-scoped binary path**: The `<platform>-<arch>` segment in `dist/bin/` prevents the universal VSIX from handing a Linux binary to macOS users. Smart design.

3. **Test coverage is excellent**: The 4 test cases cover:
   - Fallback when VS Code copy is absent (Insiders staged-install)
   - Preference for VS Code copy over bundled
   - `undefined` when nothing exists
   - Platform/arch path validation

4. **CI validation**: The `marketplace-publish.yml` change ensures the binary is actually present in the VSIX before publishing.

### Minor Suggestions

1. **VSIX size impact**: The bundled ripgrep binary adds ~20MB to the VSIX. Consider noting this in the changeset or adding a comment in `esbuild.mjs` about the size tradeoff. Users on slow connections may notice.

2. **Cross-compilation edge case**: If someone builds the VSIX on a CI runner with a different arch than the target (e.g., building on ARM64 for x86 users), the bundled binary would be wrong. This is fine for the current CI setup but worth a comment.

3. **Error message**: When `@vscode/ripgrep` doesn't provide `rgPath`, the esbuild plugin throws. Consider adding a more descriptive error: `"@vscode/ripgrep did not provide rgPath — check that the package is installed correctly"`.

### Nit

The `fileExistsAtPath` import in the test is clean and the mocking approach is idiomatic Vitest. No issues there.

**LGTM** 🚀

**Comment by 0xMink** [comment]
Superseded by #248 — it fixes the root cause directly: getBinPath now recognizes the @vscode/ripgrep-universal layout that VS Code Insiders ships, plus a system PATH fallback. Diagnostics on the affected machine showed the bug is just a path-resolution miss, so bundling a binary turned out to be unnecessary.
